vault/resource_transit_secret_backend_key.go

```diff
@@ -110,6 +110,20 @@ func transitSecretBackendKeyResource() *schema.Resource {
 					Elem: schema.TypeString,
 				},
 			},
+			"key_size": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The key size in bytes for algorithms that allow variable key sizes. Currently only applicable to HMAC, where it must be between 32 and 512 bytes.",
+				Default:     0,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(int)
+					// Default for non-hmac is 0.
+					if v == 0 || v < 32 || v > 512 {
```

Comment:
The transit engines allows setting a value of zero for non-hmac types. Maybe instead of checking for zero we should check that this is type `hmac`?

Reply:
I apologize for my ignorance - how does one get `type` from the validate function? I thought all you received was the key and value of the associated field. Is there a global object I can refer to here?

Alternately, I could remove this validation altogether and enforce the `hmac` rule as you propose elsewhere.

Reply:
Woops, sorry I was thinking of DiffSuppressFunc which allows checking arbitrary `ResourceData` fields.

I think it makes sense to remove this ValidateFunc and rely on Vault's validation.

Reply:
I think we can remove this validation from the TFVP. From [this bit of code](https://github.com/hashicorp/vault/blob/main/builtin/logical/transit/path_keys.go#L231-L239) in Vault I see that Vault is already doing extra validation on its end to determine if the `key_size` parameter is a valid value, and I think we can just let Vault handle dealing with the situation and returning a valid error. 

This way if you set a non-zero value in TFVP for a non-HMAC type, Vault will return an error. If you are of HMAC Key type and are inputting an invalid `key_size`, Vault will still return the right error. Suggest not worrying about extra validation of the same kind on the TFVP side, and we can remove the ValidateFunc from here :)

Reply:
Thanks for the clarification.